fs/devfs/DevFS.c

```
@@ -0,0 +1,56 @@
+int DevfsReadFile(const char* path, void* buffer, uint32_t max_size) {
+    // The path is the device name, e.g. "/Serial"
+    // We need to strip the leading '/'
+    const char* dev_name = path + 1;
+    CharDevice_t* dev = CharDeviceFind(dev_name);
+    if (!dev || !dev->Read) {
+        return -1;
+    }
+    return dev->Read(dev, buffer, max_size);
+}
+
+int DevfsWriteFile(const char* path, const void* buffer, uint32_t size) {
```

Clang (14.0.6)

[warning] parameter 'size' is unused

(misc-unused-parameters)